src/main/java/bwapi/Game.java

```diff
@@ -100,14 +99,18 @@
     private Text.Size textSize = Text.Size.Default;
     private boolean latcom = true;
 
+    public final SideEffectQueue sideEffects = new SideEffectQueue();
```

Comment:
sorry if I'm going over this too quickly, does this have an impact on non-async JBWAPI?

Reply:
The side effect queue is still used, but it's flushed immediately so the behavior is identical. It introduces a lambda allocation and function call per command issued, which I could remove with some additional work.

Reply:
It would really surprise me if commands issued would ever be the bottleneck of `jbwapi`'s side, could you maybe just add what you wrote as a comment above the variable for future reference?

Reply:
It just occurred to me that if latcom fabricates game state based on commands issued in the current frame, then this implementation breaks latcom. Will address.

Reply:
Async doesn't play nice with latcom in general out of the box. This needs some more attention, even if the initial solution is "don't allow latcom with async"

Reply:
We can just make latcom and async mutually exclusive, e.g. by enabling async, latcom gets disabled.

We can also add a "withLatCom(boolean enabled)" flag to the BWConfig, same for UserInput. So we don't need to call these in onStart anymore